### PR TITLE
Conditionally build Codecov container

### PR DESCRIPTION
**Pull request by christophermaier**
No need to build and push this every single time.

### CODE REVIEW

File: .buildkite/pipeline.merge.yml [hunk 1]
```diff
@@ -4,17 +4,22 @@ env:
...
-      - docker-login#v2.0.1:
-          username: grapl-cicd
-          password-env: CLOUDSMITH_API_KEY
-          server: docker.cloudsmith.io
-    agents:
-      queue: "docker"
+      - chronotc/monorepo-diff#v2.0.4:
+          diff: .buildkite/shared/scripts/diff.sh
+          log_level: "debug"
+          watch:
```
Comment (twunderlich-grapl) [inline .buildkite/pipeline.merge.yml:12]: This is pretty cool

File: .buildkite/shared/pants.toml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+[GLOBAL]
+pants_version = "2.7.1"
+backend_packages = [
+    "pants.backend.shell",
+    "pants.backend.shell.lint.shellcheck",
+    "pants.backend.shell.lint.shfmt"
+]
+
+pants_ignore = [
+    "!.buildkite/"
```
Comment (twunderlich-grapl) [inline .buildkite/shared/pants.toml:10]: Is there a reason to set this? tests and hooks both contain bash code that could benefit from shellcheck and shfmt

**Review by twunderlich-grapl — APPROVED** [review]
One non-blocking comment

File: .buildkite/shared/pants.toml (hunk 2 above)
Comment (christophermaier) [reply .buildkite/shared/pants.toml:10]: This particular one is from the git subtree, so it doesn't really "count". However, it's in the top-level `pants.toml` file, and it probably shouldn't be. I think it came in via copy/paste initially, when it really shouldn't have.

Thanks for catching this; probably need to remove it from our other plugins, too

**Pull request merged**